aws_lambda_builders/workflows/go_modules/validator.py

```diff
@@ -0,0 +1,63 @@
+"""
+Go Runtime Validation
+"""
+
+import logging
+import os
+import subprocess
+
+from aws_lambda_builders.exceptions import MisMatchRuntimeError
+
+LOG = logging.getLogger(__name__)
+
+
+class GoRuntimeValidator(object):
+    SUPPORTED_RUNTIMES = {
+        "go1.x"
+    }
+
+    def __init__(self, runtime):
+        self.language = "go"
+        self.runtime = runtime
+        self._valid_runtime_path = None
+
+    def has_runtime(self):
+        """
+        Checks if the runtime is supported.
+        :param string runtime: Runtime to check
+        :return bool: True, if the runtime is supported.
+        """
+        return self.runtime in self.SUPPORTED_RUNTIMES
+
+    def validate(self, runtime_path):
+        """
+        Checks if the language supplied matches the required lambda runtime
+        :param string runtime_path: runtime to check eg: /usr/bin/go
+        :raises MisMatchRuntimeError: Version mismatch of the language vs the required runtime
+        """
+        if not self.has_runtime():
+            LOG.warning("'%s' runtime is not "
+                        "a supported runtime", self.runtime)
+            return None
+
+        expected_major_version = self.runtime.replace(self.language, "").split('.')[0]
+
+        p = subprocess.Popen([runtime_path, "version"],
+                             cwd=os.getcwd(),
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, _ = p.communicate()
+
+        mismatched = p.returncode != 0 \
+            or len(out.split()) < 3 \
```

Comment:
Where `out = "go version go1.10.3 darwin/amd64"`, the `len(out.split()) < 3` expression will return `False`, but that's expected since this is a chained or expression where any truthy part would set `mismatched` to `True` and raise an exception later. I can definitely make this clearer to avoid confusion.

Reply:
Ahh this was a guard for getting the version out of the `go version` command. I didn't realize this initially.